app/src/components/ActionsPanel.js

```diff
@@ -1,137 +1,157 @@
             css={`
-              margin-bottom: ${3 * GU}px;
+              display: flex;
+              flex-shrink: 0;
+              align-items: center;
             `}
           >
-            <Info>{this.props.info}</Info>
-          </div>
-          <Field
-            label={
-              this.props.action == 'Wrap' ? 'Amount' : 'Wrapped token amount'
-            }
-          >
-            <TextInput
-              ref={amount => (this.amountInput = amount)}
-              value={amount}
-              min={0}
-              max={300}
-              onChange={this.handleAmountChange}
-              adornment={
-                this.props.action == 'Wrap'
-                  ? this.props.erc20TokenSymbol
-                  : this.props.wrappedTokenSymbol
-              }
-              adornmentPosition="end"
-              adornmentSettings={{
-                width: 55,
-                padding: 8,
-              }}
-              required
-              wide
+            <img
+              src={wrap}
+              css={`
+                margin: 0 ${2 * GU}px;
+              `}
             />
-          </Field>
-          <Field
-            label={
-              this.props.action == 'Wrap' ? 'Wrapped token amount' : 'Amount'
-            }
-          >
-            <TextInput
-              ref={amount => (this.amountInput = amount)}
-              value={amount}
-              onChange={this.handleAmountChange}
-              adornment={
-                this.props.action == 'Wrap'
-                  ? this.props.wrappedTokenSymbol
-                  : this.props.erc20TokenSymbol
-              }
-              adornmentPosition="end"
-              adornmentSettings={{
-                width: 55,
-                padding: 8,
-              }}
-              required
-              wide
-            />
-          </Field>
-          <Button disabled={!amount} mode="strong" type="submit" wide>
-            {this.props.action + ' tokens'}
-          </Button>
-        </form>
-      </div>
-    )
-  }
+            <span
+              css={`
+                color: ${theme.surfaceContentSecondary};
+                margin-right: ${0.5 * GU}px;
+                min-width: ${6 * GU}px;
+                max-width: ${12 * GU}px;
+                overflow: hidden;
+                text-overflow: ellipsis;
+                text-align: right;
+              `}
+            >
+              {amount || 0}
+            </span>
+            <span
+              css={`
+                color: ${theme.surfaceContentSecondary};
+              `}
+            >
+              {action === 'Wrap' ? wrappedToken.symbol : outsideToken.symbol}
+            </span>
+          </div>
+        </div>
+      </Field>
+      <Button disabled={!amount} mode="strong" type="submit" wide>
+        {action} tokens
+      </Button>
+    </form>
+  )
+}
+WrapTokensPanelContent.defaultProps = {
+  onAction: () => {},
 }
 
 export default WrapTokensPanel
```

Comment:
The panel has been re-done, to avoid two input fields that are directly connected with each other. It now just displays the converted amount beside the input field:

<img width="444" alt="Screen Shot 2019-12-03 at 3 59 42 PM" src="https://user-images.githubusercontent.com/4166642/70064638-270e4a80-15ea-11ea-9aec-9fa9c34bf72e.png">